src/envoy/http/authn/authenticator_base.cc

```diff
@@ -43,7 +43,7 @@ bool isRuleMatchedWithPeer(const iaapi::CredentialRule& rule,
 AuthenticatorBase::AuthenticatorBase(
     FilterContext* filter_context,
     const AuthenticatorBase::DoneCallback& done_callback)
-    : filter_context_(filter_context), done_callback_(done_callback) {}
+    : filter_context_(&filter_context), done_callback_(done_callback) {}
```

Comment:
what does this "&" do?  passing address of filter_context

I think the function argument should use FilterContext&

Reply:
I'd like to keep the style that mutable object is passed by pointer. I can change to ref to, if you think envoy is more consistent that way.

Reply:
This is not a style issue.

filter_context is FilterContext* type

and here you assign  filter_context_(&filter_context).  so it is address of address.